File: src/roco_serverchan_notifier/push_provider_senders/bark.py
from __future__ import annotations

from ..push_http import HttpSession, JsonPostRequest, post_json
from ..push_models import NotificationMessage, ProviderConfig, PushResult
from .common import provider_config_text


def send_bark(
    provider: ProviderConfig, message: NotificationMessage, session: HttpSession, timeout: int
) -> PushResult:
    server_url = provider_config_text(provider, "server_url").rstrip("/")
    url = f"{server_url}/{provider.config['device_key']}"
    payload = {
        "title": message.title,
        "body": f"{message.body}\n\n{message.markdown}",
    }
    group = provider_config_text(provider, "group")
    if group:
        payload["group"] = group
    return post_json(
        JsonPostRequest(
            provider,
            session,
            url,
            payload,
            timeout,
            success_codes={200, "200", 0, "0"},
        )
    )

File: cloudflare-worker/src/push-provider-senders/bark.ts
import type { NotificationMessage, ProviderConfig, PushResult } from "../types";
import { postJson } from "../push-http";
import { providerConfigText } from "./common";

export async function sendBark(
  provider: ProviderConfig,
  message: NotificationMessage,
  timeoutSec: number
): Promise<PushResult> {
  const serverUrl = providerConfigText(provider, "server_url").replace(/\/$/, "");
  const url = `${serverUrl}/${provider.config.device_key}`;
  const payload: Record<string, unknown> = {
    title: message.title,
    body: `${message.body}\n\n${message.markdown}`,
  };
  const group = providerConfigText(provider, "group");
  if (group) payload.group = group;
  return postJson(provider, url, payload, timeoutSec, {
    successCodes: new Set([200, "200", 0, "0"]),
  });
}
